src/main/java/de/zalando/aruha/nakadi/partitioning/OrderingKeyFieldsPartitioningStrategy.java

```diff
@@ -0,0 +1,71 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONObject;
+
+import java.util.List;
+
+import static java.lang.Math.abs;
+
+public class OrderingKeyFieldsPartitioningStrategy implements PartitioningStrategy {
+
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final int numberOfPartitions) throws InvalidOrderingKeyFieldsException {
```

Comment:
Yes, I was thinking about exactly this. Ideally you should pass over the available partition names to the strategy as well. I then decided that for now it only makes it more complicated. For sure, this doesn't work for messaging brokers that don't provide simple increasing numbers for their partitions, but this is a step ahead.

With having the PartitionCache implemented (see discussion about the number of partitions handed over to this method in the Controller), we can use the cache, to cache the partition names. I will refactor in a second step.